### PR TITLE
*(ticdc): remove scheduler v1

### LINKED ISSUE #5361: Remove TiCDC scheduler v1
Scheduler V1 is no longer used.

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #5361

### What is changed and how it works?

* Remove scheduler v1
* Remove TaskStatus usage in owner/processor (no longer used)
* Clean up TaskPosition usage in owner/processor (only kept `RunningError`)

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

 - Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/model/owner.go [hunk 1]
```diff
@@ -208,12 +212,12 @@ func (i *TableReplicaInfo) Clone() *TableReplicaInfo {
 	return &clone
 }
 
-// TaskStatus records the task information of a capture
+// TaskStatus records the task information of a capture.
+//
+// Deprecated: only used in API. TODO: remove API usage.
```
Comment (3AceShowHand) [inline cdc/model/owner.go:217]: It looks the whole `TaskStatus` is labeled as `deprecated`.

It has 3 different usage:
1. for old scheduler to track scheduling information.
2. for new ticdc to keep compatible with the old version.
3. for OpenAPI, return table's information.

I think the first one usage is deprecated, the last 2 is still useful.

File: cdc/model/owner.go [hunk 2]
```diff
@@ -388,7 +303,6 @@ func (status *ChangeFeedStatus) Unmarshal(data []byte) error {
 
 // ProcInfoSnap holds most important replication information of a processor
 type ProcInfoSnap struct {
```
Comment (3AceShowHand) [inline cdc/model/owner.go:305]: should `Tables` bne returned or not?

File: cdc/model/owner.go [hunk 3]
```diff
@@ -388,7 +303,6 @@ func (status *ChangeFeedStatus) Unmarshal(data []byte) error {
...
-	CfID      ChangeFeedID                  `json:"changefeed-id"`
-	CaptureID string                        `json:"capture-id"`
-	Tables    map[TableID]*TableReplicaInfo `json:"-"`
```
Comment (3AceShowHand) [inline cdc/model/owner.go:393]: It looks that `Tables` here is useless before? Have you ever test this?

**Comment by 3AceShowHand** [comment]
can we change the signature of `scheduler.Tick`, Just like `scheduler_dispatcher`.
```
Tick(
		ctx context.Context,
		checkpointTs model.Ts
		currentTables []model.TableID,
		captures map[model.CaptureID]*model.CaptureInfo,
	) (newCheckpointTs, newResolvedTs model.Ts, err error)
```

File: cdc/model/owner.go (hunk 1 above)
Comment (overvenus) [reply cdc/model/owner.go:217]: > 2. for new ticdc to keep compatible with the old version.

I don't think so, we only need to decode keys left by old version, not values.

> 3. for OpenAPI, return table's information.

We will replace them with other structs defined in http_model.go later.

File: cdc/model/owner.go (hunk 3 above)
Comment (overvenus) [reply cdc/model/owner.go:393]: No usage in ticdc code base.

**Comment by overvenus** [comment]
> can we change the signature of `scheduler.Tick`, Just like `scheduler_dispatcher`.
> 
> ```
> Tick(
> 		ctx context.Context,
> 		checkpointTs model.Ts
> 		currentTables []model.TableID,
> 		captures map[model.CaptureID]*model.CaptureInfo,
> 	) (newCheckpointTs, newResolvedTs model.Ts, err error)
> ```

Good idea, I'd like to change it in a later PR.

**Review by sdojjy — APPROVED** [review]

**Review by liuzix — APPROVED** [review]

**Pull request merged**